components/GameScreen.vue

```diff
@@ -1,10 +1,17 @@
 <script setup lang="ts">
 import { Application, Graphics, Text, FillGradient, Assets, type Texture, Sprite } from "pixi.js";
 
+const { data: user } = await useFetch("/api/auth/user");
 const refreshIntervalMs = 1000;
-const zoomSpeed = 0.05;
+const zoomSpeed = 0.1;
 const minZoom = 1;
 const maxZoom = 3;
+let startZoomTime: number;
+const zoomDuration = 500;
+let targetScale: number;
+let targetPos: { x: number; y: number };
+let isZooming = false;
+let isFollowing = false;
```

Comment:
If it's a `ref` Vue component will rerender on its changes.

```suggestion
const isFollowing = ref<boolean>(false);
```